### PR TITLE
Yet another in(secure) patch

### PR DESCRIPTION
**Pull request by miminar**
Do not consider all additional registries as secure. Check the list
of insecure registries before marking additional registry as secure.
This covers expecially the case of registries running on localhost.

Signed-off-by: Michal Minar <miminar@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
I think we should consider all Registries as secure, and force customers to identify insecure-registries.  Or is this doing that?  IE If I --add-registry foobar.com --insecure-registry=foobar.com

**Comment by miminar** [comment]
The case you mention works - `foobar.com` will be considered insecure (fallback to port 80 will work).

The cases that currently fail and should succeed (at least upstream succeeds):
- `--add-registry localhost:5000` unless `--insecure-registry localhost:5000` is also passed
- `--add-registry 127.0.0.1:5000` unless `--insecure-registry 127.0.0.1:5000` is also passed
- `--add-registry foobar.com:5000 --insecure-registry foobar.com`

In other words additional registry must match exactly some entry in insecured registry list to be considered insecure.
